realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -278,7 +278,10 @@ void BaseRealSenseNode::registerDynamicOption(ros::NodeHandle& nh, rs2::options
             {
                 sensor.set_option(option, option_value);
             }
-            ddynrec->add(new DDBool(option_name, i, sensor.get_option_description(option), option_value));
+            ddynrec->registerVariable<bool>(
+              option_name, option_value,
+              [option, sensor](bool new_value) { sensor.set_option(option, new_value); },
```

Comment:
Probably not important, but maybe capture by reference here and in the other lambdas. Here it'd be: `[&option, &sensor]`

Reply:
I believe that when option and sensor go out of scope, the references will no longer be valid, and accessing them will cause undefined behavior.

Reply:
True, these lambdas are used outside of this method scope.
I was also concerned about copying `sensor`, but it looks like this method already copies it, so it should be fine. :+1: